ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/SourceDescription.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.client;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Metadata for a ksqlDB stream or table.
+ */
+public interface SourceDescription {
+
+  /**
+   * @return name of this stream/table
+   */
+  String name();
```

Comment:
These method names (`name()`, `type()`, etc.) are a departure from the method names in other Java client interfaces (e.g., https://github.com/confluentinc/ksql/blob/master/ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/QueryInfo.java) in that they aren't prefixed with `get` (`getName()`, `getType()`, etc.). I think it's cleaner to not have the `get` but if we value consistency I can rename these.